django/db/models/functions/json.py

```diff
@@ -0,0 +1,196 @@
+            new_source_expression.extend(
+                (
+                    Value(key_paths_join),
+                    Cast(
+                        Value(value, output_field=self.output_field),
+                        output_field=self.output_field,
+                    ),
+                )
+            )
+
+        copy.set_source_expressions(new_source_expression)
+
+        return super(JSONSet, copy).as_sql(
+            compiler,
+            connection,
+            function="JSON_SET",
+            **extra_context,
+        )
+
+    def as_postgresql(self, compiler, connection, **extra_context):
+        copy = self.copy()
+
+        all_items = list(self.fields.items())
+        key, value = all_items[0]
+        rest = all_items[1:]
+
+        if rest:
+            copy.fields = {key: value}
+            return JSONSet(copy, **dict(rest)).as_postgresql(
+                compiler, connection, **extra_context
+            )
+        else:
+            new_source_expression = copy.get_source_expressions()
+            key_paths = key.split(LOOKUP_SEP)
+            key_paths_join = ",".join(key_paths)
+            new_source_expression.extend(
+                (
+                    Value(f"{{{key_paths_join}}}"),
+                    Value(value, output_field=self.output_field),
+                )
+            )
+            copy.set_source_expressions(new_source_expression)
+        return super(JSONSet, copy).as_sql(
+            compiler, connection, function="JSONB_SET", **extra_context
+        )
+
+    def as_oracle(self, compiler, connection, **extra_context):
+        copy = self.copy()
+
+        all_items = list(self.fields.items())
+        key, value = all_items[0]
+        rest = all_items[1:]
+
+        if rest:
+            copy.fields = {key: value}
+            return JSONSet(copy, **dict(rest)).as_oracle(
+                compiler, connection, **extra_context
+            )
+        else:
+            new_source_expression = copy.get_source_expressions()
+            key_paths = key.split(LOOKUP_SEP)
+            key_paths_join = compile_json_path(key_paths)
+            new_source_expression.extend(
+                (Value(value, output_field=self.output_field),)
+            )
+            copy.set_source_expressions(new_source_expression)
+
+        class ArgJoiner:
+            def join(self, args):
+                if len(args) < 2:
+                    return ", ".join(args)
+                else:
+                    return f"{args[0]}, SET '{key_paths_join}' = {args[-1]} FORMAT JSON"
+
+        return super(JSONSet, copy).as_sql(
+            compiler,
+            connection,
+            function="JSON_TRANSFORM",
+            arg_joiner=ArgJoiner(),
+            **extra_context,
+        )
+
+
+class JSONRemove(Func):
+    def __init__(self, field_name, *paths):
+        if not paths:
+            raise TypeError("JSONRemove requires paths parameter")
```

Comment:
```suggestion
        if not paths:
            raise TypeError("JSONRemove requires at least one path to remove")
```

Also, I think we should add a test case for this to ensure it's covered, e.g. something like

```py
with self.assertRaisesMessage(TypeError, "JSONRemove requires at least one path to remove"):
    JSONRemove("some_json")
```